BCI_analysis/io/io_suite2p.py

```diff
@@ -1,11 +1,198 @@
+    data_dict = BCI_analysis.io_python.read_sessionwise_npy(file_path)
+    all_trial_frame_nums = data_dict['all_si_frame_nums']
+    all_trial_names=data_dict['all_si_filenames']
+    trial_names_with_event = data_dict['closed_loop_filenames']
+    trial_event_times = data_dict['go_cue_times']
+    frame_rate = data_dict['sampling_rate']
+    event_indices = trial_times_to_session_indices(all_trial_frame_nums,
+                                                   all_trial_names,
+                                                   trial_names_with_event,
+                                                   trial_event_times,
+                                                   frame_rate)
+    """
+    all_trial_start_frames = np.asarray([0] + np.cumsum(all_trial_frame_nums))[:-1]
+    
+    event_indices = []
+    for trial_i, (trial_name,event_times) in enumerate(zip(trial_names_with_event,trial_event_times)):
+        trial_start_idx = all_trial_start_frames[np.where(trial_name == np.asarray(all_trial_names))[0][0]]
+        print(trial_start_idx)
+        for t in event_times:
+            event_indices.append(trial_start_idx+int(t*frame_rate))
+    return event_indices
+            
+        
+def align_trace_to_event(F,
+                         event_indices,
+                         frames_before,
+                         frames_after):
+    """
+    Creates an event-locked array of traces with given size
+
+    Parameters
+    ----------
+    F : matrix of float
+        sessionwise fluorescence
+    event_indices : list of int
+        indices of events (output of trial_times_to_session_indices )
+    frames_before : int
+        number of frames to keep before event
+    frames_after : int
+        number of frames to keep after event.
+
+    Returns
+    -------
+    F_aligned : matrix of float (frames * cells * trials)
+        trial-locked matrix
+
+    """
+    max_frames= frames_before+frames_after
+    F_aligned = np.ones((max_frames, F.shape[0], len(event_indices)))*np.nan
+    
+    for i, center_idx in enumerate(event_indices):
+
+        start_frame = center_idx - frames_before
+        end_frame = center_idx + frames_after
+        if end_frame > F.shape[1]:
+            end_frame = F.shape[1]
+        if end_frame - start_frame > frames_after:
+            end_frame = start_frame + frames_after
+        start_frame = start_frame - frames_before # taking 40 time points before trial starts
+        if start_frame<0: # taking care of edge at the beginning
+            missing_frames_at_beginning = np.abs(start_frame)
+            start_frame = 0
+        else:
+            missing_frames_at_beginning = 0
+        F_aligned[missing_frames_at_beginning:missing_frames_at_beginning+end_frame-start_frame, :, i] = F[:, start_frame:end_frame].T
+    return F_aligned
+
+    
+
+
+def sessionwise_to_trialwise(F, 
```

Comment:
does this function do something other than the top 2 combined? Or does this do the same thing as `align_trace_to_event`